gef-ui/server/controllers/about-facility/index.js

```diff
@@ -365,21 +147,22 @@ const validateAboutFacility = async (req, res) => {
       dealId,
       facilityId,
       status,
-      isFacilityEndDateFeatureFlagEnabled: isFacilityEndDateFeatureFlagEnabled(),
+      isFacilityEndDateEnabled: isFacilityEndDateEnabledOnDeal(deal.version ?? 0),
+      facilityEndDateExists,
     });
   }
 
   try {
-    const deal = await api.getApplication({ dealId, userToken });
     await api.updateFacility({
       facilityId,
       payload: {
-        name: body.facilityName,
-        shouldCoverStartOnSubmission: isTrueSet(body.shouldCoverStartOnSubmission),
-        monthsOfCover: body.monthsOfCover || null,
+        name: facilityName,
+        shouldCoverStartOnSubmission: isTrueSet(shouldCoverStartOnSubmission),
+        monthsOfCover: monthsOfCover || null,
         coverStartDate: coverStartDate ? format(coverStartDate, DATE_FORMAT.COVER) : null,
         coverEndDate: coverEndDate ? format(coverEndDate, DATE_FORMAT.COVER) : null,
         coverDateConfirmed: deal.submissionType === DEAL_SUBMISSION_TYPE.AIN ? true : null,
+        facilityEndDateExists: isFacilityEndDateEnabledOnDeal(deal.version ?? 0) ? isTrueSet(facilityEndDateExists) : undefined,
```

Comment:
I think because of how the forms work, these will always be strings (if set) - maybe it would make more sense to throw an error if this goes wrong but hopefully the validation would catch this